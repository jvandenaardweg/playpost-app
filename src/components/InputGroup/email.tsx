import React from 'react'
import { TextInput, TextInputProps } from 'react-native'

import textInput from '../../constants/text-input';
import { InputGroup } from './index';

export interface Props extends TextInputProps {
  label?: string;
  textInputRef?: React.RefObject<TextInput>
}

export const InputGroupEmail: React.FC<Props> = React.memo(({ label, textInputRef, ...rest }) => {
  return (
    <InputGroup label={label ? label : 'E-mail address'}>
      <TextInput
        ref={textInputRef}
        autoCapitalize="none"
        textContentType="username"
        keyboardType="email-address"
        placeholder="john.appleseed@playpost.app"
        style={textInput}
        clearButtonMode="always"
        blurOnSubmit={false}
        {...rest}
      />
    </InputGroup>
  )
})
