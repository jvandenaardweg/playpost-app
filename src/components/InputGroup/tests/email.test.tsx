import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { InputGroupEmail, Props } from '../email';

const defaultProps: Props = { }

describe('InputGroupEmail', () => {
  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }
      wrapper = render(<InputGroupEmail {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should render correctly render the correct default label', () => {
      expect(wrapper.getByTestId('InputGroup-Text-label').props.children).toBe('E-mail address');
    });

    it('should render correctly render the correct custom label', () => {
      const props = {
        ...defaultProps,
        label: 'Your new e-mail address'
      }

      wrapper.update(<InputGroupEmail {...props} />);

      expect(wrapper.getByTestId('InputGroup-Text-label').props.children).toBe('Your new e-mail address');
    });
  });
});
