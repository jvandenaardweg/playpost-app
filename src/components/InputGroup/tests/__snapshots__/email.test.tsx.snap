// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroupEmail rendering should render correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 14,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#090909",
          "fontFamily": "Inter-SemiBold",
          "fontSize": 15,
          "lineHeight": 23,
        },
        Object {
          "marginBottom": 4,
        },
      ]
    }
    testID="InputGroup-Text-label"
  >
    E-mail address
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    blurOnSubmit={false}
    clearButtonMode="always"
    keyboardType="email-address"
    placeholder="john.appleseed@playpost.app"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#EEEEEE",
        "borderRadius": 6,
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "height": 50,
        "lineHeight": undefined,
        "paddingLeft": 14,
        "paddingRight": 14,
      }
    }
    textContentType="username"
    underlineColorAndroid="transparent"
  />
</View>
`;
